### ISSUE TITLE
Did not generate the brain mask.

### ISSUE THREAD

**Issue by alenchen-0606** [ecalabr/brain_mask#1]
Hello, thanks for your good work. 
I want to try the following command: 
python brain_mask.py -m t1-t1c-t2-flair -i example_data 
The command window show as follow:
![messageImage_1631382606825](https://user-images.githubusercontent.com/85331604/132956788-224a487d-dd4c-4325-b754-8dadf8888159.jpg)

But i can't find brain mask in example_data\example001.
(I delete the example001_brain_mask.nii.gz first.)

**Comment by ecalabr** [comment]
Not sure what's going on here but I suspect it's a bug related to windows
filepaths since I have only tested in linux and darwin. Can you try using
the -o flag and specify a complete output directory path? If that works
then I can probably figure out how to fix this bug.

thanks!
-Evan

**Comment by ecalabr** [comment]
One other idea is to ensure that you can successfully run tensorflow without errors, and also try to run the script with GPU disabled (-f). Tensorflow and GPU related errors sometimes don't get sent to terminal output.

**Comment by alenchen-0606** [comment]
Thanks for your quick reply.

Do you mean try the following command?
python Brain_mask.py -m t1-t1c-t2-flair -i D:\brain_mask\example_data -o D:\brain_mask\example_data -f
![messageImage_1631385933174](https://user-images.githubusercontent.com/85331604/132958252-604f7b3d-0083-44b3-ac4d-5de2ea0c91b3.jpg)

It also did not generate brain mask.
Sorry, i don't know how to ensure that run tensorflow without errors.

But thank you for your response!!

**Comment by ecalabr** [comment]
Try changing lines 13 and 14 in brain_mask.py to:

os.environ['TF_CPP_MIN_LOG_LEVEL'] = '0' 
logging.getLogger('tensorflow').setLevel(logging.INFO)

Then re-run. This should allow all tensorflow output to go to the terminal and may help us identify whats going wrong. If that doesn't work then I will have to try testing the code on a windows system.

**Comment by alenchen-0606** [comment]
Rerun as you requested, i get the brain mask. (But the mask is no as same as you provide.)
![error](https://user-images.githubusercontent.com/85331604/132969943-6a8fddc5-401d-4bd6-ab2c-f3435ffb1c72.jpg)

And i try rerun again i get the feedback as follow:
![error2](https://user-images.githubusercontent.com/85331604/132969973-98c88aed-40c3-4981-a5c1-f60760ad010b.jpg)

Only the first time i rerun i can get brain mask.

The WARNING:
WARNING:tensorflow:Calling GradientTape.gradient on a persistent tape inside its context is significantly less efficient than calling it outside the context (it causes the gradient ops to be recorded on the tape, leading to increased CPU and memory usage). Only call GradientTape.gradient inside the context if you actually want to trace the gradient in order to compute higher order derivatives.

**Comment by ecalabr** [comment]
I just uploaded a new commit that should help. It fixes the tensorflow warning and adds better logging. I think it is failing because it is finding a probability file from a previous run that won't be overwritten. Try the latest commit and try using the -x flag to overwrite existing data. At the very least this should give more detailed logs.

Note that your output may not be the same as the mask I have included, and may not look correct. The sample data is already masked (for patient confidentiality purposes) and is included just as a test to see if the script runs. Masking will not work well on data that is already masked. If you want to assess performance, you will have to supply your own data.

Hope this helps!

**Comment by alenchen-0606** [comment]
Sorry for the late reply.
I can't find a new commit that you mentioned.
Where can i find it?

**Comment by ecalabr** [comment]
My apologies! I had forgotten that github is no longer allowing password authentication, so my push failed. The latest commit is now available. Please let me know how it goes.

**Comment by alenchen-0606** [comment]
I try to run the new commit, i get following error:
![error](https://user-images.githubusercontent.com/85331604/133557320-972ad3cc-f8c5-46b8-9fd3-b1f7902660aa.jpg)
I think the first half warns that CUDA does not install the correct version, so it can only use cpu.
But I don’t know the meaning of the AssertionError in the second half.

I'm not sure if it is my personal problem
Sorry for bothering you

**Comment by ecalabr** [comment]
Yes - CUDA did not install correctly. Unfortunately this can be a difficult issue to fix. I can suggest trying to reinstall from instructions or searching stack overflow for any errors you are finding. However, you should still be able to use the code in CPU mode.

As for the assertion error, it seems the script is not able to find any of the pre-trained models in the "trained_models" directory, which should be in the same place as the brain_mask.py script. Please make sure this directory is present and that it contains subdirectories with pre-trained models. If there is an issue, you could try deleting and re-cloning the entire codebase.

**Comment by alenchen-0606** [comment]
I delete a '/' in brain_mask.py line 161. 
![image](https://user-images.githubusercontent.com/85331604/133640333-caee241b-990d-41d3-bef5-deb76418d2b9.png)
So the trained model can be read.
And i can get a mask on my own data.
Thank you for your help these day.

**Comment by ecalabr** [comment]
Great! I updated this part of the code to be more compatible. I am closing this issue for now. Good luck!
